configure.ac

```diff
@@ -1475,7 +1475,22 @@ AC_ARG_ENABLE(omkafka,
          esac],
         [enable_omkafka=no]
 )
+AC_ARG_ENABLE(kafka_tests,
+        [AS_HELP_STRING([--enable-kafka-tests],[enable Kafka specific tests in testbench @<:@default=no@:>@])],
+        [case "${enableval}" in
+         yes) enable_kafka_tests="yes" ;;
+          no) enable_kafka_tests="no" ;;
+           *) AC_MSG_ERROR(bad value ${enableval} for --enable-kafka-tests) ;;
+         esac],
+        [enable_kafka_tests=no]
+)
+AM_CONDITIONAL(ENABLE_KAFKA_TESTS, test x$enable_kafka_tests = xyes)
+
 if test "x$enable_omkafka" = "xyes"; then
+	if test "x$enable_kafka_tests" = "xyes"; then
+		AX_PROG_JAVAC #we don't need javac, but macro documentation says JAVAC *must* be checked before JAVA
+		AX_PROG_JAVA
+	fi
```

Comment:
no,no,no all is well. My fault.

2016-04-19 18:07 GMT+02:00 Janmejay Singh <notifications@github.com>:

> In configure.ac
> <https://github.com/rsyslog/rsyslog/pull/967#discussion_r60260659>:
>
> >  if test "x$enable_omkafka" = "xyes"; then
> > +	if test "x$enable_kafka_tests" = "xyes"; then
> > +		AX_PROG_JAVAC #we don't need javac, but macro documentation says JAVAC *must* be checked before JAVA
> > +		AX_PROG_JAVA
> > +	fi
>
> The other option is to not check it in configure, but fail the test when
> it can't find JVM.
>
> I can switch it to that if you prefer.
>
> Or we can change the help-text "enable Kafka specific tests in testbench"
> to "enable Kafka specific tests in testbench (requires JVM)".
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/rsyslog/pull/967/files/003b8a80ff4c37a4c1b93e34a670c225980b170d#r60260659>
>